routers/admin/users.go

```diff
@@ -267,7 +267,10 @@ func EditUserPost(ctx *context.Context, form auth.AdminEditUserForm) {
 			ctx.ServerError("UpdateUser", err)
 			return
 		}
-		u.HashPassword(form.Password)
+		if err = u.SetPassword(form.Password); err != nil {
+			ctx.InternalServerError(err)
```

Comment:
`ServerError` is the correct function but not `InternalServerError`

Reply:
@lunny this should be not issue - It's used else where too:
* https://github.com/go-gitea/gitea/pull/14282/files#diff-e9dab0c829326d553cbf4a8c95d2f55d7b2541317aec8018b9e247d79352bfd9L285
* https://github.com/go-gitea/gitea/pull/14282/files#diff-e9dab0c829326d553cbf4a8c95d2f55d7b2541317aec8018b9e247d79352bfd9R293

and is only a shortcout for ServerError ...

Reply:
@lunny but here it is: https://github.com/go-gitea/gitea/pull/14341